x/rollapp/keeper/rollapp.go

```diff
@@ -48,6 +51,120 @@ func (k Keeper) RegisterRollapp(ctx sdk.Context, rollapp types.Rollapp) error {
 	return nil
 }
 
+func (k Keeper) UpdateRollapp(ctx sdk.Context, update *types.UpdateRollappInformation) error {
+	if err := update.ValidateBasic(); err != nil {
+		return fmt.Errorf("validate update: %w", err)
+	}
+
+	updated, err := k.checkIfCanUpdateRollapp(ctx, update)
+	if err != nil {
+		return err
+	}
+
+	k.SetRollapp(ctx, updated)
+
+	// Emit event
+	if err = ctx.EventManager().EmitTypedEvent(&updated); err != nil {
+		return fmt.Errorf("emit event: %w", err)
+	}
+
+	return nil
+}
+
+func (k Keeper) checkIfCanUpdateRollapp(ctx sdk.Context, update *types.UpdateRollappInformation) (current types.Rollapp, err error) {
+	var found bool
+	current, found = k.GetRollapp(ctx, update.RollappId)
+	if !found {
+		err = errRollappNotFound
+		return
+	}
+
+	if update.Creator != current.Creator {
+		err = sdkerrors.ErrUnauthorized
+		return
+	}
+
+	if current.Frozen {
+		err = types.ErrRollappFrozen
+		return
+	}
+
+	_, hasState := k.GetLatestStateInfoIndex(ctx, update.RollappId)
```

Comment:
probably should check if some update requested to immutable field
and return if `hasState`